test/dialect/mysql/test_reflection.py

```diff
@@ -375,13 +377,147 @@ def test_reflection_with_table_options(self):
             assert reflected.kwargs["mysql_default charset"] == "utf8"
             assert reflected.kwargs["mysql_avg_row_length"] == "3"
             assert reflected.kwargs["mysql_connection"] == "fish"
+            assert reflected.kwargs["mysql_stats_sample_pages"] == "4"
 
             # This field doesn't seem to be returned by mysql itself.
             # assert reflected.kwargs['mysql_password'] == 'secret'
 
             # This is explicitly ignored when reflecting schema.
             # assert reflected.kwargs['mysql_auto_increment'] == '5'
 
+    def test_reflection_with_partition_options(self):
+        kwargs = dict(
+            mysql_engine="InnoDB",
+            mysql_default_charset="utf8",
+            mysql_partition_by="HASH(MONTH(c2))",
+            mysql_partitions="6",
+        )
+
+        def_table = Table(
+            "mysql_def",
+            MetaData(testing.db),
+            Column("c1", Integer()),
+            Column("c2", DateTime),
+            **kwargs
+        )
+
+        def_table.create()
+        try:
+            reflected = Table("mysql_def", MetaData(testing.db), autoload=True)
+        finally:
+            def_table.drop()
+
+        assert def_table.kwargs["mysql_partition_by"] == "HASH(MONTH(c2))"
+        assert def_table.kwargs["mysql_partitions"] == "6"
+
+        assert reflected.kwargs["mysql_partition_by"] == "HASH (month(`c2`))"
+        assert reflected.kwargs["mysql_partitions"] == "6"
+
+    def test_reflection_with_subpartition_options(self):
+
+        subpartititon_text = """HASH (TO_DAYS (c2))
+                                SUBPARTITIONS 2(
+                                 PARTITION p0 VALUES LESS THAN (1990),
+                                 PARTITION p1 VALUES LESS THAN (2000),
+                                 PARTITION p2 VALUES LESS THAN MAXVALUE
+                             );"""
+        kwargs = dict(
+            mysql_engine="InnoDB",
+            mysql_default_charset="utf8",
+            mysql_partition_by="RANGE(YEAR(c2))",
+            mysql_subpartition_by=subpartititon_text,
+        )
```

Comment:
**mike bayer** (zzzeek) wrote:

use eq_(a, b) for all the "assert == " here

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/2185

Reply:
**Ramon Williams** (RamonWill) wrote:

Done

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/2185